scripts/showcase/samplesheet-prep-test.py

```diff
@@ -227,12 +239,38 @@ def convert_sample_sheet_header_to_v2(sample_sheet_header_rows):
     A list of header rows with modified content as per V2_METADATA_COLUMN_CHANGES
     """
     sample_sheet_header_rows_new = []
-    for row in sample_sheet_header_rows:
+
+    # Get reads
+    reads_index = sample_sheet_header_rows.index("[Reads]\n")
+    settings_index = sample_sheet_header_rows.index("[Settings]\n")
+
+    for row_index, row in enumerate(sample_sheet_header_rows):
         # Initialise new row
         new_row = row
         for old_value, new_value in V2_METADATA_COLUMN_CHANGES.items():
             new_row = re.sub(old_value + ",", new_value + ",", new_row)
+
+        if row_index == settings_index + 1 and override_cycles:
+            # Lets sneak in an overridecycles row
+            r1_mask = "{}y".format(int(sample_sheet_header_rows[reads_index+1]))
+            r2_mask = "{}y".format(int(sample_sheet_header_rows[reads_index+2]))
+            if not index_len == index_len_orig:
+                # We need to mask the length of the difference
+                i7_mask = "{}i{}n".format(index_len, index_len_orig-index_len)
+            else:
+                # Just append the length of the index
+                i7_mask = "{}i".format(index_len)
+            if not index2_len == index2_len_orig:
+                # We need to mask the length of the difference
+                i5_mask = "{}i{}n".format(index2_len, index2_len_orig-index2_len)
+            else:
+                i5_mask = "{}i".format(index2_len)
+
+            sample_sheet_header_rows_new.append("OverrideCycles,{};{};{};{}\n".format(
+                r1_mask, i7_mask, i5_mask, r2_mask
+            ))
         sample_sheet_header_rows_new.append(new_row)
```

Comment:
An example of a correct format for overridecycles setting for bclConvert is:
`OverrideCycles,Y151;I8N2;I8N2;Y151`

You are adding:
OverrideCycles,151y;8i2n;8i2n;151y

Reply:
:facepalm:

Reply:
This should be fixed now with #57